c/datasetService.c

```diff
@@ -137,6 +140,56 @@ static int serveDatasetContents(HttpService *service, HttpResponse *response){
   return 0;
 }
 
+/* new for ENQ */
+
+static int serveDatasetEnqueue(HttpService *service, HttpResponse *response){
+  zowelog(NULL, LOG_COMP_ID_MVD_SERVER, ZOWE_LOG_DEBUG2, "begin %s\n", __FUNCTION__);
+
+  zowelog(NULL, LOG_COMP_ID_MVD_SERVER, ZOWE_LOG_DEBUG2, "%s\n", __FUNCTION__);
+
+  HttpRequest *request = response->request;
+
+  if (!strcmp(request->method, methodGET)) {
+    char *l1 = stringListPrint(request->parsedFile, 1, 1, "/", 0);
+    char *percentDecoded = cleanURLParamValue(response->slh, l1);
+    char *filenamep1 = stringConcatenate(response->slh, "//'", percentDecoded);
+    char *filename = stringConcatenate(response->slh, filenamep1, "'");
+    zowelog(NULL, LOG_COMP_ID_MVD_SERVER, ZOWE_LOG_DEBUG, "Taking enqueue on: %s\n", filename);
+    fflush(stdout);
+    respondWithEnqueue(response, filename, TRUE);
+  }
+  else if (!strcmp(request->method, methodDELETE)) {
+    char *l1 = stringListPrint(request->parsedFile, 1, 1, "/", 0);
+    char *percentDecoded = cleanURLParamValue(response->slh, l1);
+    char *filenamep1 = stringConcatenate(response->slh, "//'", percentDecoded);
+    char *filename = stringConcatenate(response->slh, filenamep1, "'");
```

Comment:
Can any of these be NULL?

Reply:
Interesting question and good spot of a potential error case.  

These 4 lines are used by `serveDatasetContents`, `serveDatasetEnqueue` (my new function) and `serveVSAMDatasetContents`, when the logic requires the DSN to be wrapped in the format `//'dsn'` for use by USS.  If either of my 2 usages of these 4 lines produces an error, then so could the other 4 uses in `serveDatasetContents` and  `serveVSAMDatasetContents`.  

This error can only happen with the API, since the file editor always ensures that the DSN is not null.  I don't know if the API will catch that before it gets to this C program.  

I'd like @struga0258 to create a test for this case.  If that test fails, then I suggest we create a new issue and fix it at that stage.

Reply:
unchanged from original, need to review again, so far didnt encounter any error here, will ned to check edge cases like sending empty filename etc